### PR TITLE
Never use raw arguments when running on Mono

### PR DESCRIPTION
**Pull request by marek-safar**
Fixes issues #9954 and #11317

### CODE REVIEW

File: src/Compilers/Shared/BuildClient.cs [hunk 1]
```diff
@@ -171,14 +171,25 @@ public Task<RunCompilationResult> RunCompilationAsync(IEnumerable<string> origin
...
-            if (IsRunningOnWindows)
+            if (UseNativeArguments())
             {
                 return GetCommandLineWindows(args);
             }
 
             return args;
         }
 
+        static bool UseNativeArguments()
```
Comment (jaredpar) [inline src/Compilers/Shared/BuildClient.cs]: Please use explicit accessibility modifiers here.

File: src/Compilers/Shared/BuildClient.cs [hunk 2]
```diff
@@ -171,14 +171,25 @@ public Task<RunCompilationResult> RunCompilationAsync(IEnumerable<string> origin
...
+        {
+            if (!IsRunningOnWindows)
+                return false;
```
Comment (jaredpar) [inline src/Compilers/Shared/BuildClient.cs]: Please use braces even on single line ifs.

**Comment by marek-safar** [comment]
Updated based on the feedback

**Comment by jaredpar** [comment]
CC @dotnet/roslyn-compiler for a second review. 
@agocke

**Comment by jcouv** [comment]
@jaredpar I went ahead and merged since you already reviewed.

**Pull request merged**